fmpz/test/t-xgcd_minimal.c

```diff
@@ -0,0 +1,627 @@
+/*
+    Copyright (C) 2021 Albin Ahlbäck
+
+    This file is part of FLINT.
+
+    FLINT is free software: you can redistribute it and/or modify it under
+    the terms of the GNU Lesser General Public License (LGPL) as published
+    by the Free Software Foundation; either version 2.1 of the License, or
+    (at your option) any later version.  See <https://www.gnu.org/licenses/>.
+*/
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <gmp.h>
+#include "flint.h"
+#include "ulong_extras.h"
+#include "fmpz.h"
+
+int
+main(void)
+{
+    int i, j, result;
+    fmpz_t maxval;
+    fmpz_t d, x, y, a, b;
+    fmpz_t tx, ty;
+
+    FLINT_TEST_INIT(state);
+    fmpz_set_str(maxval, "999999999999999999999999999999999999999999", 10);
+
+    flint_printf("xgcd_minimal....");
+    fflush(stdout);
+
+
+    fmpz_init(d);
+    fmpz_init(x);
+    fmpz_init(y);
+    fmpz_init(a);
+    fmpz_init(b);
+
+    fmpz_xgcd_minimal(d, x, y, a, b); /* check small (0, 0)
+                                       * gives (0, 0, 0) */
+    result = (fmpz_is_zero(d)
+           && fmpz_is_zero(x)
+           && fmpz_is_zero(y)
+           && fmpz_is_zero(a)
+           && fmpz_is_zero(b));
+
+    if (!result)
+    {
+        flint_printf("FAIL:\n\n");
+        flint_printf("d = "), fmpz_print(d), flint_printf("\n");
+        flint_printf("x = "), fmpz_print(x), flint_printf("\n");
+        flint_printf("y = "), fmpz_print(y), flint_printf("\n");
+        flint_printf("a = "), fmpz_print(a), flint_printf("\n");
+        flint_printf("b = "), fmpz_print(b), flint_printf("\n");
+        abort();
+    }
+
+
+    _fmpz_promote_val(a);
```

Comment:
There's no need to test promoted values work. There's no expectation that any fmpz function accept promoted fmpz's. Promoted values are for internal use.